src/backend/youtube/confirmYoutube.ts

```diff
@@ -0,0 +1,156 @@
+import { StatusCodes } from 'http-status-codes';
+import { z } from 'zod';
+import {
+  Request,
+  ResponseToolkit,
+  ResponseObject,
+  ServerRoute,
+} from '@hapi/hapi';
+
+import got from 'got';
+
+import Boom from '@hapi/boom';
+
+import { Claim } from '@kiltprotocol/core';
+
+import {
+  deleteSecret,
+  getSessionBySecret,
+  getSessionWithDid,
+  setSession,
+} from '../utilities/sessionStorage';
+import { paths } from '../endpoints/paths';
+import { configuration } from '../utilities/configuration';
+
+import { youtubeEndpoints } from './youtubeEndpoints';
+import { youtubeCType } from './youtubeCType';
+
+const zodPayload = z.object({
+  sessionId: z.string(),
+  code: z.string(),
+  secret: z.string(),
+});
+
+export type Input = z.infer<typeof zodPayload>;
+
+export interface Output {
+  channelName: string;
+  channelId: string;
+}
+
+async function handler(
+  request: Request,
+  h: ResponseToolkit,
+): Promise<ResponseObject> {
+  const { logger } = request;
+  logger.debug('Youtube authorization started');
+
+  const { secret, sessionId, code } = request.payload as Input;
+
+  // This is the initial session in the first tab the user has open
+  const firstSession = getSessionBySecret(secret);
+  if (!firstSession) {
+    throw Boom.notFound('No session found for secret');
+  }
+  logger.debug('Found session with secret');
+  deleteSecret(secret);
+
+  const session = getSessionWithDid({ sessionId });
+
+  logger.debug('Exchanging code for access token');
+  const body = (await got
+    .post(youtubeEndpoints.token, {
+      form: {
+        code: decodeURIComponent(code),
+        grant_type: 'authorization_code',
+        client_id: configuration.youtube.clientId,
+        client_secret: configuration.youtube.clientSecret,
+        redirect_uri: youtubeEndpoints.redirectUri,
+      },
+    })
+    .json()) as { access_token: string };
+  logger.debug('Access token granted, fetching youtube profile');
+
+  const headers = {
+    authorization: `Bearer ${body.access_token}`,
+  };
+
+  const searchParams = {
+    part: 'id,snippet',
+    mine: 'true',
+  };
+
+  const channelListResponse = (await got(youtubeEndpoints.profile, {
+    headers,
+    searchParams,
+  }).json()) as { items?: Array<{ id: string; snippet: { title: string } }> };
+
+  const { items } = channelListResponse;
```

Comment:
What do you think about this alternative?
```suggestion
  const { items } = (await got(youtubeEndpoints.profile, {
    headers,
    searchParams,
  }).json()) as { items?: Array<{ id: string; snippet: { title: string } }> };
```